frontend/components/filter/Carousel/index.tsx

```diff
@@ -37,7 +37,23 @@ const Carousel: FunctionComponent<Props> = (props) => {
             <FlatList
                 style={styles.list}
                 data={props.choices}
-                renderItem={({ item, index }) => <ChoiceComponent choice={item} onPress={(event) => index === props.selectedChoice ? props.onChange(-1) : props.onChange(index)} selected={props.selectedChoice === index} />}
+                renderItem={({ item, index }) => <ChoiceComponent 
+                                                    choice={item} 
+                                                    onPress={
+                                                        (event) => { 
+                                                            if (item.isGenre && ((props.selectedChoice & (1 << index)) !== 0)) {

```

Comment:
passes in negative index if a genre was already selected and needs to be unselected